### PR TITLE
Add component ID to API key identifier

### PR DESCRIPTION
**Pull request by mevan-karu**
### Purpose
$subject 

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by ThumulaPerera — APPROVED** [review]

**Review by VirajSalaka — APPROVED** [review]

File: enforcer-parent/enforcer/src/test/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticatorTest.java [hunk 1]
```diff
@@ -100,6 +101,7 @@ public void retrieveTokenFromRequestCtxTest_cached_validKey() throws APISecurity
         RequestContext.Builder requestContextBuilder = new RequestContext.Builder("/api-key");
         requestContextBuilder.matchedAPI(new APIConfig.Builder("Petstore")
                 .basePath("/test")
+                .uuid("6003a3b7-af0f-4fb3-853e-a6562b2345f2")
```
[the remark this answers is not in the document]
Comment (mevan-karu) [reply enforcer-parent/enforcer/src/test/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticatorTest.java:110]: This is not used in the tests. Will remove these in a separate PR

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20241106.37

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : dev-deployment-v2 : 20241106.37

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20241106.37

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20241107.1

**Comment by choreo-cicd** [comment]
[]  : dev-deployment-v2 : 20241107.1

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : dev-deployment-v2 : 20241107.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20241107.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20241107.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20241107.4

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20241109.1

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20241109.1

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20241109.1

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20241109.1

**Pull request merged**